test/DockerSdk.Tests/scripts/docker-compose.yml

```diff
@@ -0,0 +1,18 @@
+﻿# Sets up the docker containers needed by the project's tests.
```

Comment:
Have you tried to do this by using different containers? So you don't require to run compose, and you could even run tests on a remote machine

Reply:
I hadn't considered a remote machine. That's an interesting idea!

It would definitely be nicer to containerize the testing, but I ran into some troubles and decided to give it up for now. I think it's a little beyond my current Dockerization skill level. I'd be pleased if someone more familiar with containerized tests would revamp this at some point, though. In particular, if we could get some Docker-in-Docker going, we could test against different daemon versions/configuration.

The main problem I had is that I don't want to give up the ability to debug tests from Visual Studio. VS has the ability to debug a _program_ in a container, but as far as I could tell it doesn't run _tests_ in a container.

Reply:
@Emdot I could help you on that, if you ever want to run tests on a dockerized environment, but first I need your help.
I hit a wall on #487 
Can't cancel a stream.Read() call once it is reading the stream, because inside  the stream reader it is using stream.read() and there is no cancellation token there.

And that's why they are using stream.Dispose() to cancel a stream read operation. This is what i suspect but is not confirmed. 
The truth is tests aren't working because cancellation token is ignored